src/NHibernate/Loader/Loader.cs

```diff
@@ -980,22 +981,63 @@ private void CheckVersion(int i, IEntityPersister persister, object id, object e
 			for (int i = 0; i < cols; i++)
 			{
 				object obj = null;
+				ILoadable persister = persisters[i];
+
 				EntityKey key = keys[i];
 
 				if (key == null)
 				{
-					// do nothing
-					/* TODO NH-1001 : if (persisters[i]...EntityType) is an OneToMany or a ManyToOne and
-					 * the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
-					 * an load with "outer join" the relation can be considerer loaded even if the key is null (mean not found)
+					if (cols == 1 || CollectionOwners == null)
+					{
+						continue;
+					}
+
+					int position = GetPosition(i);
+
+					if (position == -1)
+					{
+						continue;
+					}
+					
+					/*
+					* the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
+					* an load with "outer join" the relation can be considerer loaded even if the key is null (mean not found)
 					*/
+					
+					
+					foreach (int owner in CollectionOwners)
```

Comment:
`CollectionOwners` is not empty in the current tests because Employee has a "Phones" collection.